Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -146,8 +155,6 @@ void FunctionDefinitionWriter::visit(AST::StructDecl& structDecl)
         IndentationScope scope(m_indent);
         for (auto& member : structDecl.members()) {
             m_stringBuilder.append(m_indent);
-            if (m_structRole == AST::StructRole::ArgumentBuffer)
```

Comment:
Are we sure we want ArgumentBuffer to be a member of Role? Doesn't it kind of leak MSL concepts into the AST? Shouldn't we be saying "BindGroup" instead?

Maybe leaking MSL concepts into the AST is a lost cause? 😕

Reply:
I think this is also a side effect of the fact that I'm still learning how to translate those terms. I'm also not 100% we can avoid the leak, but it doesn't seem strictly necessary for now.